consumer/pdata/trace_test.go

```diff
@@ -139,5 +139,5 @@ func TestResourceSpansWireCompatibility(t *testing.T) {
 
 	// Now compare that the original and final ProtoBuf messages are the same.
 	// This proves that goproto and gogoproto marshaling/unmarshaling are wire compatible.
-	assert.True(t, gogoproto.Equal(*pdataRS.orig, &gogoprotoRS2))
+	assert.EqualValues(t, *pdataRS.orig, &gogoprotoRS2)
```

Comment:
gogoproto.Equal does not work correctly with the custom data type. I had to use assert.EqualValues which we don't normally use for Protobufs because it may not work if the data is not laid out deterministically. Luckily in our case we don't have maps or unknown fields so the data is deterministic and comparison works. I am not if it is worth spending more time on this right now, if for some reason this does not prove to be reliable I can look into it in the future.